Dashboard/app/js/lib/views/folder-survey-selector.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import ChildOption from 'akvo-flow/components/ChildOption';
+import observe from '../mixins/observe';
+
+require('akvo-flow/views/react-component');
+
+FLOW.FolderSurveySelectorView = FLOW.ReactComponentView.extend(observe({
+  // 'FLOW.projectControl.content.isLoaded': 'folderSurveySelector',
+}), {
+  init() {
+    this._super();
+    this.handleChange = this.handleChange.bind(this);
+    this.folderSurveySelector = this.folderSurveySelector.bind(this);
+    this.state = { value: 0 };
+    this.surveyGroups = [];
+  },
+
+  didInsertElement(...args) {
+    this._super(...args);
+    if (FLOW.projectControl.content.isLoaded) {
+      FLOW.projectControl.get('content').forEach((item) => {
+        this.surveyGroups.push({
+          keyId: item.get('keyId'),
+          parentId: item.get('parentId'),
+          name: item.get('name'),
+          published: item.get('published'),
+          projectType: item.get('projectType'),
+          monitoringGroup: item.get('monitoringGroup'),
+          ancestorIds: item.get('ancestorIds'),
+        });
+      });
+      this.folderSurveySelector(0);
+    }
+  },
+
+  folderSurveySelector(parentId) {
+    if (!FLOW.projectControl.content.isLoaded) return;
+    const levels = [];
+    let surveySelected = false;
+    // if not root level, generate ancestor selectors
+    if (parentId !== 0) {
+      const parent = this.surveyGroups.find(sg => sg.keyId == parentId);
+      for (let i = 0; i < parent.ancestorIds.length; i++) {
+        const level = [{
```

Comment:
Is it possible to use a placeholder like we have always been using, instead of an object whose function mimics the placeholder?

Reply:
It is possible but the idea being to move away from Ember means we currently mimic the Ember objects we've been using to avoid calling functions such as get when accessing data. This way the components such as the `ChildOption` I've created can be reused going forward by simply passing an object to them